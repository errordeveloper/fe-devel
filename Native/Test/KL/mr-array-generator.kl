/*
 *  Copyright 2010-2012 Fabric Engine Inc. All rights reserved.
 */

operator generator1(
  io Integer value
  )
{
  report("generator1: value=" + value);
  value = 42;
}

operator generator2(
  io Integer value,
  Index index
  )
{
  report("generator2: value=" + value + " index=" + index);
  value = 2*index;
}

operator generator3(
  io Integer value,
  Index index,
  Size count
  )
{
  report("generator3: value=" + value + " index=" + index + " count=" + count);
  value = 2*index;
}

operator generator4(
  io Integer value,
  Index index,
  Size count,
  String shared
  )
{
  report("generator4: value=" + value + " index=" + index + " count=" + count + " shared=" + shared);
  value = 2*index;
}

operator entry() {
  ValueProducer<Size> cvg = createConstValue(Size(10));

  ArrayProducer<Integer> ag1 = createArrayGenerator(cvg, generator1);
  report(ag1);
  report(ag1.getCount());
  for (Index i=0; i<10; ++i)
    report(ag1.produce(i));

  ArrayProducer<Integer> ag2 = createArrayGenerator(cvg, generator2);
  report(ag2);
  report(ag2.getCount());
  for (Index i=0; i<10; ++i)
    report(ag2.produce(i));

  ArrayProducer<Integer> ag3 = createArrayGenerator(cvg, generator3);
  report(ag3);
  report(ag3.getCount());
  for (Index i=0; i<10; ++i)
    report(ag3.produce(i));

  ArrayProducer<Integer> ag4 = createArrayGenerator(cvg, generator4, createConstValue("hello"));
  report(ag4);
  report(ag4.getCount());
  for (Index i=0; i<10; ++i)
    report(ag4.produce(i));
}
