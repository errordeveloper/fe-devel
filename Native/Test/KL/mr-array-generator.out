ArrayProducer<Integer>
10
generator1: value=0
42
generator1: value=0
42
generator1: value=0
42
generator1: value=0
42
generator1: value=0
42
generator1: value=0
42
generator1: value=0
42
generator1: value=0
42
generator1: value=0
42
generator1: value=0
42
ArrayProducer<Integer>
10
generator2: value=0 index=0
0
generator2: value=0 index=1
2
generator2: value=0 index=2
4
generator2: value=0 index=3
6
generator2: value=0 index=4
8
generator2: value=0 index=5
10
generator2: value=0 index=6
12
generator2: value=0 index=7
14
generator2: value=0 index=8
16
generator2: value=0 index=9
18
ArrayProducer<Integer>
10
generator3: value=0 index=0 count=10
0
generator3: value=0 index=1 count=10
2
generator3: value=0 index=2 count=10
4
generator3: value=0 index=3 count=10
6
generator3: value=0 index=4 count=10
8
generator3: value=0 index=5 count=10
10
generator3: value=0 index=6 count=10
12
generator3: value=0 index=7 count=10
14
generator3: value=0 index=8 count=10
16
generator3: value=0 index=9 count=10
18
ArrayProducer<Integer>
10
generator4: value=0 index=0 count=10 shared=hello
0
generator4: value=0 index=1 count=10 shared=hello
2
generator4: value=0 index=2 count=10 shared=hello
4
generator4: value=0 index=3 count=10 shared=hello
6
generator4: value=0 index=4 count=10 shared=hello
8
generator4: value=0 index=5 count=10 shared=hello
10
generator4: value=0 index=6 count=10 shared=hello
12
generator4: value=0 index=7 count=10 shared=hello
14
generator4: value=0 index=8 count=10 shared=hello
16
generator4: value=0 index=9 count=10 shared=hello
18
